### PR TITLE
Delete operation in headers policy

### LINKED ISSUE #922: Add 'delete' operation in the headers policy [THREESCALE-1354]
The APIcast Headers Policy allows to add/set/push header values, but currently, there isn't a "delete" operation.

Setting an empty value works as the delete operation in APIcast. However, the 3scale UI doesn't allow for an empty value.

Ref: https://issues.jboss.org/browse/THREESCALE-1354

### PR DESCRIPTION
**Pull request by davidor**
Closes #922

### CODE REVIEW

File: spec/policy/headers/headers_spec.lua [hunk 1]
```diff
@@ -70,6 +96,31 @@ describe('Headers policy', function()
...
+            local headers_policy = HeadersPolicy.new(config)
+
+            headers_policy:rewrite()
+
+            assert.is_nil(ngx.req.get_headers()[header])
+          end)
+        end)
+
+        describe('and the header is already set', function()
+          it('deletes it', function()
```
Comment (davidor) [inline spec/policy/headers/headers_spec.lua]: I wasn't totally sure about this one but decided to keep it working as it is now.

File: gateway/src/apicast/policy/headers/apicast-policy.json [hunk 2]
```diff
@@ -40,7 +44,7 @@
               "type": "string"
             },
             "value": {
-              "description": "Value that will be added, set or pushed in the header",
+              "description": "Value that will be added, set or pushed in the header. Not needed when deleting.",
```
[the remark this answers is not in the document]
Comment (mikz) [reply gateway/src/apicast/policy/headers/apicast-policy.json]: I think this is fine. I think showing extra field is better than adding a whole new array just for deletes.

[the remark this answers is not in the document]
Comment (davidor) [reply gateway/src/apicast/policy/headers/apicast-policy.json]: :+1:

[the remark this answers is not in the document]
Comment (mayorova) [reply gateway/src/apicast/policy/headers/apicast-policy.json]: Yeah, I agree. As long as it's clear that `value` is not required for deleting, it's OK.

**Comment by pestanko** [comment]
@davidor This one is not in the DR1 right?

**Comment by mikz** [comment]
@pestanko nope, https://github.com/3scale/apicast/compare/3scale-2.4.0-DR1...master

**Comment by pestanko** [comment]
@mikz I have tested this on nightly, so far this works.

**Pull request merged**